ooniapi/services/ooniprobe/src/ooniprobe/download_geoip.py

```diff
@@ -0,0 +1,146 @@
+"""
+Updates asn.mmdb and cc.mmdb in /var/lib/ooniapi/
+
+"""
+
+import sys
+import gzip
+import timeit
+import shutil
+import logging
+
+import geoip2.database
+from pathlib import Path
+from datetime import datetime, timezone
+from urllib.error import HTTPError
+from urllib.request import urlopen, Request
+
+from prometheus_client import metrics
+
+
+class Metrics:
+    GEOIP_ASN_NODE_CNT = metrics.Gauge("geoip_asn_node_cnt", "Count of geoi nodes")
+    GEOIP_ASN_EPOCH = metrics.Gauge("geoip_asn_epoch", "Geoip current ASN epoch")
+    GEOIP_CC_NODE_CNT = metrics.Gauge("geoip_cc_node_cnt", "Geoip asn node count")
+    GEOIP_CC_EPOCH = metrics.Gauge("geoip_cc_epoch", "Geoip current CC epoch")
+    GEOIP_CHECKFAIL = metrics.Counter(
+        "ooni_geoip_checkfail", "How many times did the check fail in geo ip fail"
+    )
+    GEOIP_UPDATED = metrics.Counter(
+        "ooni_geoip_updated", "How many times was the geoip database updated"
+    )
+    GEOIP_DOWNLOAD_TIME = metrics.Histogram(
+        "geoip_download_time", "How long it takes to download the DB"
+    )
+
+
+TS = datetime.now(timezone.utc).strftime("%Y-%m")
```

Comment:
I think you have to place this TS calculation inside of the function body, otherwise it will be evaluated only upon module import which in the case of the period operation might be only once for the lifecycle of the application.

That is to say that it might not get updated to the next month, unless the the whole module gets re-imported, which I am not sure is the behavior of the period task scheduler.